File: android/src/main/java/org/ligi/passandroid/model/PassStore.java
package org.ligi.passandroid.model;

import android.support.annotation.Nullable;
import android.content.Context;

import java.util.List;

public interface PassStore {

    void deleteCacheForId(String id);

    void refreshPassesList();

    @Nullable
    Pass getPassbookForId(String id);

    boolean deletePassWithId(String id);

    String getPathForID(String id);

    void preCachePassesList();

    List<FiledPass> getPassList();

    @Nullable
    Pass getCurrentPass();

    void setCurrentPass(@Nullable Pass pass);

    PassClassifier getClassifier(Context context);
}

File: android/src/main/java/org/ligi/passandroid/model/PassStore.kt
package org.ligi.passandroid.model

import org.ligi.passandroid.model.pass.Pass
import java.io.File

interface PassStore {

    fun save(pass: Pass)

    fun getPassbookForId(id: String): Pass?

    fun deletePassWithId(id: String): Boolean

    fun getPathForID(id: String): File

    val passMap: Map<String, Pass>

    var currentPass: Pass?

    val classifier: PassClassifier

    fun notifyChange()

    fun syncPassStoreWithClassifier(defaultTopic: String)
}
